test/extended/cli/builds.go

```diff
@@ -0,0 +1,179 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+const (
+	bcOutputTemplate      = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	testDockerFileContent = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
+)
+
+type expectedBuild struct {
+	bcName      string
+	isName      string
+	sourceType  string
+	outputType  string
+	outputImage string
+}
+
+func (r *expectedBuild) setDefaults() {
+	if r.outputType == "ImageStreamTag" && r.isName == "" {
+		r.isName = r.bcName
+	}
+}
+
+func testNewBuild(oc *exutil.CLI, opts expectedBuild) {
+	opts.setDefaults()
+	// new build
+	buildOut, err := oc.Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(buildOut).To(o.And(
+		o.ContainSubstring(fmt.Sprintf("buildconfig.build.openshift.io \"%v\" created", opts.bcName)),
+		o.ContainSubstring("Success"),
+	))
+	if opts.sourceType == "Dockerfile" {
+		o.Expect(buildOut).To(o.ContainSubstring(fmt.Sprintf("imagestream.image.openshift.io \"tests\" created"))) // see testDockerFileContent
+	}
+	if opts.isName != "" {
+		o.Expect(buildOut).To(o.ContainSubstring(fmt.Sprintf("imagestream.image.openshift.io \"%v\" created", opts.isName)))
+	}
+	// check source type
+	out, err := oc.Run("get").Args("bc/" + opts.bcName).Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(out).To(o.ContainSubstring(opts.sourceType))
+	// check correct output type and image
+	out, err = oc.Run("get").Args("bc/"+opts.bcName, "--template", bcOutputTemplate).Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	if opts.outputType == "" {
+		o.Expect(out).To(o.BeEmpty())
+	} else {
+		o.Expect(out).To(o.Equal(fmt.Sprintf("%v %v", opts.outputType, opts.outputImage)))
+	}
+	// cleanup
+	resources := []string{"bc/" + opts.bcName}
```

Comment:
The cleanup can be moved to `g.AfterEach()`

Reply:
I am not sure if I should have AfterEach since I still haven't ported all the other tests and they will have different resources.

Or should  I just call `oc delete all --all` for simplicity there?

Reply:
I have not researched the framework properly before. It seems it handles the deletion of the namespace in `AfterEach`, so additional logic should not be needed.